### ISSUE TITLE
Add support fo jdk9 and jdk10

### ISSUE THREAD

**Issue by javadev** [aNNiMON/Lightweight-Stream-API#157]
It may be new settings for travis-ci.

**Comment by PromanSEW** [comment]
Please add these new methods from Java 10:

> A new method orElseThrow has been added to the Optional class. It is synonymous with and is now the preferred alternative to the existing get method.

https://docs.oracle.com/javase/10/docs/api/java/util/Optional.html#orElseThrow()

> New methods toUnmodifiableList, toUnmodifiableSet, and toUnmodifiableMap have been added to the Collectors class in the Stream package. These allow the elements of a Stream to be collected into an unmodifiable collection.

https://docs.oracle.com/javase/10/docs/api/java/util/stream/Collectors.html#toUnmodifiableList() and other `toUnmodifiable*` collectors

Source:
http://www.oracle.com/technetwork/java/javase/10-relnote-issues-4108729.html#NewFeature

**Comment by aNNiMON** [comment]
@PromanSEW since `toUnmodifiable*` collectors use Java 9 `List.of`, the preferred alternative in LSA would be 

```java
Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList);
```

Unfortunately, `Collections.unmodifiable*` methods are added in Android API 24. I can't drop Android pre-24 support.

**Comment by PromanSEW** [comment]
@aNNiMON I have just checked, these methods are exist in API 16 at least. I think they existed since older API than 16 too. Android documentation can contain errors, for example `ConcurrentLinkedDeque` is added in API 21, but I found this class in API 16 too. I think that checking source code of earlier APIs is good idea.

**Comment by aNNiMON** [comment]
@PromanSEW you're right. I just check Android 2.3 source code and it has [`Collections.unmodifiableList`](https://android.googlesource.com/platform/libcore/+/android-2.3_r1/luni/src/main/java/java/util/Collections.java#2363). It's bad that Android documentation contains errors like this.

**Comment by javadev** [comment]
oraclejdk10 is also supported by travis-ci now.
